velox/functions/prestosql/aggregates/ApproxDistinctAggregate.cpp

```diff
@@ -150,6 +150,18 @@ class ApproxDistinctAggregate : public exec::Aggregate {
     return false;
   }
 
+  bool supportsToIntermediate() const final {
+    return hllAsRawInput_;
+  }
+
+  void toIntermediate(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      VectorPtr& result) const final {
+    VELOX_CHECK_EQ(args.size(), 1);
+    result->copy(args[0].get(), rows, nullptr);
```

Comment:
Nice. We can probably avoid a copy as well.

Reply:
@Yuhta Jimmy, I just realized that we might need to handle masked out rows. See example in velox/functions/prestosql/aggregates/MinMaxAggregates.cpp

```
  void toIntermediate(
      const SelectivityVector& rows,
      std::vector<VectorPtr>& args,
      VectorPtr& result) const override {
    const auto& input = args[0];
    if (rows.isAllSelected()) {
      result = input;
      return;
    }

    auto* pool = BaseAggregate::allocator_->pool();

    result = BaseVector::create(input->type(), rows.size(), pool);

    // Set result to NULL for rows that are masked out.
    {
      BufferPtr nulls = allocateNulls(rows.size(), pool, bits::kNull);
      rows.clearNulls(nulls);
      result->setNulls(nulls);
    }

    result->copy(input.get(), rows, nullptr);
  }
```